toolkit/tools/roast/formats/squashfs.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package formats
+
+import (
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/shell"
+)
+
+// SquashFSType represents the squashfs format
+const SquashFSType = "squashfs"
+
+// SquashFS implements Converter interface to convert a RAW image into a squashfs file
+type SquashFS struct {
+}
+
+// Convert converts the image in the squashfs format
+func (t *SquashFS) Convert(input, output string, isInputFile bool) (err error) {
+	const squashErrors = false
+	err = shell.ExecuteLive(squashErrors, "mksquashfs", input, output, "-noappend")
```

Comment:
Can you remind me -- are there cases in which this will convert a .tar directly into a .squashfs? If so, then do we need to specify something appropriate for `-root-mode` since .tar files don't actually capture the root directory's properties? Or are we now only ever capturing from an expanded out directory?

Reply:
It's not converting a tar to squashfs, and yes it captures an expanded directory.